src/ConfigProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * @link      http://github.com/zendframework/zend-serializer for the canonical source repository
+ * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Serializer;
+
+class ConfigProvider
+{
+    /**
+     * Return configuration for this component.
+     *
+     * @return array
+     */
+    public function __invoke()
+    {
+        return [
+            'dependencies' => $this->getDependencyConfig(),
+        ];
+    }
+
+    /**
+     * Return dependency mappings for this component.
+     *
+     * @return array
+     */
+    public function getDependencyConfig()
+    {
+        return [
+            'factories' => [
+                'SerializerAdapterManager' => AdapterPluginManagerFactory::class,
```

Comment:
It's a magic string, and kept for BC purposes with existing ZF2 applications. It could potentially be pushed to an alias, but that presents a new issue: if you register a factory under the alias name, the alias resolves first, which means the override factory won't be used.

Definitely considered using FQCNs for these, but the migration implications get pretty nasty.